gate-web/src/main/groovy/com/netflix/spinnaker/gate/services/CredentialsService.groovy

```diff
@@ -29,21 +28,33 @@ class CredentialsService {
   @Autowired
   ClouddriverService clouddriverService
 
-  List<ClouddriverService.Account> getAccounts() {
+  Collection<String> getAccountNames() {
+    getAccountNames([])
+  }
+
+  Collection<String> getAccountNames(Collection<String> userRoles) {
+    getAccounts(userRoles)*.name
+  }
+
+  Collection<ClouddriverService.Account> getAccounts() {
+    getAccounts([])
+  }
+
+  /**
+   * Returns all account names that a user with the specified list of userRoles has access to.
+   */
+  List<ClouddriverService.Account> getAccounts(Collection<String> userRoles) {
```

Comment:
Feels like the changes to this method might warrant inclusion in the commit comment as well.